modules/phase_field/src/userobjects/DiscreteNucleationFromFile.C

```diff
@@ -103,7 +103,7 @@ DiscreteNucleationFromFile::initialize()
   _changes_made = {0, 0};
 
   // expire entries from the local nucleus list (if the current time step converged)
-  if (_fe_problem.converged())
+  if (_fe_problem.checkConvergence() > 0)
```

Comment:
Wait, also a checkConvergence() call too?

Reply:
Yes, this is the issue I raised: checking solver status is not safe inside moose object because solver status may not be valid at the point of checking.

Reply:
It's not that it's invalid, it's just that the result may not be expected. Perhaps the converged flag should be reset every time we advance timesteps? If we did that and somebody had a user object checking the flag before the solve, they would get an "unsolved or not converged" flag back and then they'd have to think a little bit about what they were trying to do. I wouldn't over engineer this. It might just be worth talking to the developers or looking at the code to understand the intent.

Reply:
Looks like only @dschwen is doing this. I will create a new enum `MOOSE_PICARD_UNSOLVED` and set solver status in `FEProblemBase::timestepSetup`.